packages/chat-component-bindings/src/utils/getParticipantsWhoHaveReadMessage.ts

```diff
@@ -7,22 +7,21 @@ import { ChatMessageWithStatus } from '@internal/chat-stateful-client';
  *logic: Looking at message A, how do we know it's read number?
  * Assumption: if user read the latest message, user has read all messages before that
  * ReadReceipt behaviour: read receipt is only sent to the last message
- * if participant read a message that is sent later than message A, then the participant has read message A
- * how do we check if the message is sent later than message A?
- * we compare if the readon timestamp is later than the message A sent time
- * if last read message id is not equal to message A's id, check the read on time stamp.
- * if the last read message is read after the message A is sent, then user should have read message A as well */
+ *
+ * If participant read a message that is sent later than message A, then the participant has read message A
+ * How do we check if the message is sent later than message A?
+ * We compare if the messageID of the last read message is larger than or equal to the message A's id
+ * Because messageID is the creation timestamp of each message
+ * if MessageId of B is larger than message Id of A, then B is created after A
+ * if the last read message is created after the message A is sent, then user should have read message A as well */
 export const getParticipantsWhoHaveReadMessage = (
   message: ChatMessageWithStatus,
-  readReceiptForEachSender: { [key: string]: { lastReadMessage: string; readOn: Date; name: string } }
+  readReceiptForEachSender: { [key: string]: { lastReadMessage: string; name: string } }
 ): { id: string; name: string }[] => {
   return (
     Object.entries(readReceiptForEachSender)
       // Filter to only read receipts that match the message OR the participant has read a different message after this message has been created
-      .filter(
-        ([_, readReceipt]) =>
-          readReceipt.lastReadMessage === message.id || new Date(readReceipt.readOn) >= new Date(message.createdOn)
-      )
+      .filter(([_, readReceipt]) => parseInt(readReceipt.lastReadMessage, 10) >= parseInt(message.id, 10))
```

Comment:
Instead of parseInt, a direct string comparing is safer, it is the same result if you compare 2 number strings:
https://javascript.info/comparison#:~:text=To%20see%20whether%20a%20string%20is%20greater%20than%20another%2C%20JavaScript%20uses%20the%20so%2Dcalled%20%E2%80%9Cdictionary%E2%80%9D%20or%20%E2%80%9Clexicographical%E2%80%9D%20order.

even the max int would be 900719925474099(very long, and would be safe enough), but the string comparation would support unlimited length

Reply:
Updated